core/arch/arm/kernel/secure_partition.c

```diff
@@ -170,17 +170,23 @@ bool sp_has_exclusive_access(struct sp_mem_map_region *mem,
 	return !sp_mem_is_shared(mem);
 }
 
-static uint16_t new_session_id(struct sp_sessions_head *open_sessions)
+static TEE_Result new_session_id(uint16_t *endpoint_id)
 {
-	struct sp_session *last = NULL;
-	uint16_t id = SPMC_ENDPOINT_ID + 1;
+	struct sp_session *session = NULL;
 
-	last = TAILQ_LAST(open_sessions, sp_sessions_head);
-	if (last)
-		id = last->endpoint_id + 1;
+	*endpoint_id = SPMC_ENDPOINT_ID;
 
-	assert(id > SPMC_ENDPOINT_ID);
-	return id;
+	/* Find the first available endpoint id */
+	do {
+		(*endpoint_id)++;
+
+		session = sp_get_session(*endpoint_id);
+
+		if (*endpoint_id <= SPMC_ENDPOINT_ID)
```

Comment:
This can only happen after `*endpoind_id` has wrapped. I would prefer testing just before `(*endpoint_id)++` against `UINT16_MAX`. It doesn't change much, but it's a bit more explicit.

Reply:
Corrected